phobos/model/models.py

```diff
@@ -1023,16 +1023,34 @@ def deriveModelDictionary(root, name='', objectlist=[]):
         # parse joint and motor information
         if sUtils.getEffectiveParent(link):
             # joint may be None if link is a root
+            # to prevent confusion links are always defining also joints
             jointdict = deriveJoint(link, logging=True, adjust=True)
             log("  Setting joint type '{}' for link.".format(jointdict['type']), 'DEBUG')
+            # first check if we have motor information in the joint properties
```

Comment:
The initial thought for moving the motor/sensor/controller definitions away from the links was to clean out possible overriding chains which are confusing and can make the model difficult to debug.

Therefore, we decided to create the motor/sensor/controller objects to have a **single** point in the model where this information can be found. We introduced model configurations to allow the user to select which elements of the model should be exported. Thus, each link can have infinite motor objects attached to it, but only the selection or saved configuration will be used.

Is there a specific reason for this change? Maybe our usecases are not complete?

Reply:
Concerning the motor export, the main reason was backwards compatibility. Many old models at RIC have their motor parameters in the links and with this change the feature of attaching motor and controller objects is still there but we don't have to adapt all our old models if we want to slightly change them, like a single parameter.
Secondly, for me it is much more simpler to debug a model if I have four motor parameters in the links then having infinite many motor and controller objects with redundant parameters connected to a link.
Additionally, the usability for me is much higher when keeping the amount of Blender objects, that I have to select on different layers and that I have to maintain, as low as possible. For complex models it can easily become an unnecessary challenge to arrange the motor and controller objects to be able to select the one you want to edit.

Reply:
I think the backwards compatibility is a valid point. I have also found the managing of multiple motor and controller objects to not really simplify the workflow compared to earlier versions, though keeping this information in separate "categories" and not hiding it in links is in principle still a good idea.

I suggest revisiting this problem once we start working on Phobos for Blender 2.8x for good and adding Malte's fix for the time being. I'm going to test and merge this code and close this pull request.

Reply:
Agreed, I didn't have the old models in mind!

We can use the Blender collections of 2.8x for this matter.